### PR TITLE
[redis, docs]: Bump redis and docs/Pillow dependencies

### LINKED ISSUE #11765: [ray] Critical security vulnerabilities in Redis/Lua dependency
### What is the problem?

- Ray version and other system information (Python version, TensorFlow version, OS):  1.x

3 Ray components have *critical* security vulnerabilities that prevent them from deploying into any enterprise or cloud infrastructures for production usage.

- c-ares
Current version - 1.14.0 OUTDATED 
Latest version  - 1.16.1
CVE-2016-5180 
CVE-2007-3152 
CVE-2007-3153 

- Lua
Current version - 5.1.5 OUTDATED 
Latest version  - 5.4.1
CVE-2020-15889
CVE-2020-24342
CVE-2020-15888
CVE-2014-5461
CVE-2020-15945

- Redis
Current version -5.0.9 OUTDATED 
Latest version - 6.0.9
CVE-2020-14147

### PR DESCRIPTION
**Pull request by barakmich**
## Why are these changes needed?

Closes https://github.com/ray-project/ray/issues/11765

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: bazel/ray_deps_setup.bzl [hunk 1]
```diff
@@ -107,8 +115,8 @@ def ray_deps_setup():
         name = "com_github_tporadowski_redis_bin",
         build_file = "//bazel:BUILD.redis",
         strip_prefix = None,
-        url = "https://github.com/tporadowski/redis/releases/download/v4.0.14.2/Redis-x64-4.0.14.2.zip",
-        sha256 = "6fac443543244c803311de5883b714a7ae3c4fa0594cad51d75b24c4ef45b353",
+        url = "https://github.com/tporadowski/redis/releases/download/v5.0.9/Redis-x64-5.0.9.zip",
+      sha256 = "b09565b22b50c505a5faa86a7e40b6683afb22f3c17c5e6a5e35fc9b7c03f4c2",
```
Comment (simon-mo) [inline bazel/ray_deps_setup.bzl:119]: indent

**Review by simon-mo — APPROVED** [review]
LGTM

**Comment by simon-mo** [comment]
The Java&Streaming failure (might?) be related to this? Just re-triggered the build.

**Comment by ericl** [comment]
cc @raulchen could we get help on the Java "start redis failure" with the upgraded redis? There don't seem to be any error logs.

**Comment by chaokunyang** [comment]
@barakmich The logs in /tmp/ray/session_latest/logs/redis.out are:
```
18481:C 11 Nov 2020 10:27:39.869 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
18481:C 11 Nov 2020 10:27:39.869 # Redis version=6.0.9, bits=64, commit=00000000, modified=0, pid=18481, just started
18481:C 11 Nov 2020 10:27:39.869 # Configuration loaded
18481:M 11 Nov 2020 10:27:39.871 # Not listening to IPv6: unsupported
18481:M 11 Nov 2020 10:27:39.872 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
18481:M 11 Nov 2020 10:27:39.872 # Server initialized
18481:M 11 Nov 2020 10:27:39.872 # Module /tmp/ray/session_2020-11-11_10-27-39_742609/libray_redis_module.so failed to load: It does not have execute permissions.
18481:M 11 Nov 2020 10:27:39.872 # Can't load module from /tmp/ray/session_2020-11-11_10-27-39_742609/libray_redis_module.so: server aborting
```
Does it relate to newer redis version? 
The redis start commands example is `Starting process redis with command: /tmp/ray/session_2020-11-11_10-36-47_150057/redis-server --protected-mode no --port 61957 --loglevel warning --loadmodule /tmp/ray/session_2020-11-11_10-36-47_150057/libray_redis_module.so --requirepass  5241590000000000`

**Comment by ericl** [comment]
@chaokunyang it seems the error is "It does not have execute permissions", any reason why this might happen in Java only?

**Comment by chaokunyang** [comment]
Try this diff, works on my machine
--- a/java/runtime/src/main/java/io/ray/runtime/runner/RunManager.java
+++ b/java/runtime/src/main/java/io/ray/runtime/runner/RunManager.java
@@ -257,6 +257,10 @@ public class RunManager {
     final File redisServerFile = BinaryFileUtil.getNativeFile(
         rayConfig.sessionDir, BinaryFileUtil.REDIS_SERVER_BINARY_NAME);
     Preconditions.checkState(redisServerFile.setExecutable(true));
+    // The redis module file.
+    File redisModule = BinaryFileUtil.getNativeFile(
+        rayConfig.sessionDir, BinaryFileUtil.REDIS_MODULE_LIBRARY_NAME);
+    Preconditions.checkState(redisModule.setExecutable(true));
     List<String> command = Lists.newArrayList(
         // The redis-server executable file.
         redisServerFile.getAbsolutePath(),
@@ -268,8 +272,7 @@ public class RunManager {
         "warning",
         "--loadmodule",
         // The redis module file.
-        BinaryFileUtil.getNativeFile(
-            rayConfig.sessionDir, BinaryFileUtil.REDIS_MODULE_LIBRARY_NAME).getAbsolutePath()
+        redisModule.getAbsolutePath()
     );

**Comment by chaokunyang** [comment]
Try this diff, works on my machine:
```
--- a/java/runtime/src/main/java/io/ray/runtime/runner/RunManager.java
+++ b/java/runtime/src/main/java/io/ray/runtime/runner/RunManager.java
@@ -257,6 +257,10 @@ public class RunManager {
     final File redisServerFile = BinaryFileUtil.getNativeFile(
         rayConfig.sessionDir, BinaryFileUtil.REDIS_SERVER_BINARY_NAME);
     Preconditions.checkState(redisServerFile.setExecutable(true));
+    // The redis module file.
+    File redisModule = BinaryFileUtil.getNativeFile(
+        rayConfig.sessionDir, BinaryFileUtil.REDIS_MODULE_LIBRARY_NAME);
+    Preconditions.checkState(redisModule.setExecutable(true));
     List<String> command = Lists.newArrayList(
         // The redis-server executable file.
         redisServerFile.getAbsolutePath(),
@@ -268,8 +272,7 @@ public class RunManager {
         "warning",
         "--loadmodule",
         // The redis module file.
-        BinaryFileUtil.getNativeFile(
-            rayConfig.sessionDir, BinaryFileUtil.REDIS_MODULE_LIBRARY_NAME).getAbsolutePath()
+        redisModule.getAbsolutePath()
     );

```

**Comment by chaokunyang** [comment]
I'm not sure whether newer redis need redis mudule to be executable. @ericl

**Comment by ericl** [comment]
Looks good! Streaming tests are known failure.

**Pull request merged**